### ISSUE TITLE
HTTP Request to /timeTable failed

### ISSUE THREAD

**Issue by DeiEFLE** [apache/incubator-kie-optaplanner-quickstarts#121]
Hi,

I had already reported this issue (see [109](https://github.com/kiegroup/optaplanner-quickstarts/issues/109)) and it was closed as resolved, but I'm arfaid I must tell you, it is not.
The behaviour stayed exactly the same as before.

My setup and steps have stayed the same.
(Appart from using the lates sources from git - also I'm not sure if anything appart from dependencies has actually changed)

> Setup:
> Win10,
> eclipse 4.19.0,
> gradle 6.8
> 
> I went with 'live coding' and started the app with the gradle command as suggested in the readme.
> When refreshing the browser the initial call to 'getTimeTable()' crashes.

It seems like a Path is created from an URI causing the error.
Please let me know, if I am doing anthing wrong, or if you have any further questions.

```
Working Directory: C:\Users\####\git\optaplanner-quickstarts\quarkus-school-timetabling
Gradle user home: C:\Users\####\.gradle
Gradle Distribution: Gradle wrapper from target build
Gradle Version: 6.8
Java Home: C:\java\jdk-11.0.2+9_win64
JVM Arguments: None
Program Arguments: None
Build Scans Enabled: false
Offline Mode Enabled: false
Gradle Tasks: quarkusDev


> Task :quarkusGenerateCode
preparing quarkus application

> Task :compileJava UP-TO-DATE
> Task :processResources UP-TO-DATE
> Task :classes UP-TO-DATE

> Task :quarkusDev
Listening for transport dt_socket at address: 5005
__  ____  __  _____   ___  __ ____  ______ 
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
2021-06-16 13:14:14,639 INFO  [io.quarkus] (Quarkus Main Thread) quarkus-school-timetabling 0.1.0-SNAPSHOT on JVM (powered by Quarkus 1.13.3.Final) started in 16.827s. Listening on: http://localhost:8080
2021-06-16 13:14:14,644 INFO  [io.quarkus] (Quarkus Main Thread) Profile dev activated. Live Coding activated.
2021-06-16 13:14:14,644 INFO  [io.quarkus] (Quarkus Main Thread) Installed features: [agroal, cdi, hibernate-orm, hibernate-orm-panache, hibernate-orm-rest-data-panache, jdbc-h2, mutiny, narayana-jta, optaplanner, optaplanner-jackson, resteasy, resteasy-jackson, smallrye-context-propagation, webjars-locator]
2021-06-16 13:20:01,390 ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (executor-thread-1) HTTP Request to /timeTable failed, error id: 986939e6-4c15-4b28-9be4-d9fef4865aff-1: org.jboss.resteasy.spi.UnhandledException: java.nio.file.InvalidPathException: Illegal char <:> at index 4: file:///C:/Users/####/.gradle/caches/modules-2/files-2.1/org.optaplanner/optaplanner-quarkus/8.7.0.Final/886edda6912bc7f4fff3a394b18794bad59a7732/optaplanner-quarkus-8.7.0.Final.jar
	at org.jboss.resteasy.core.ExceptionHandler.handleApplicationException(ExceptionHandler.java:106)
	at org.jboss.resteasy.core.ExceptionHandler.handleException(ExceptionHandler.java:372)
	at org.jboss.resteasy.core.SynchronousDispatcher.writeException(SynchronousDispatcher.java:218)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:519)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:261)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:161)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
	at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:164)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:247)
	at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:73)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:138)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.access$000(VertxRequestHandler.java:41)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler$1.run(VertxRequestHandler.java:93)
	at io.quarkus.runtime.CleanableExecutor$CleaningRunnable.run(CleanableExecutor.java:231)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2415)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at java.base/java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: java.nio.file.InvalidPathException: Illegal char <:> at index 4: file:///C:/Users/####/.gradle/caches/modules-2/files-2.1/org.optaplanner/optaplanner-quarkus/8.7.0.Final/886edda6912bc7f4fff3a394b18794bad59a7732/optaplanner-quarkus-8.7.0.Final.jar
	at java.base/sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:182)
	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:153)
	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
	at java.base/sun.nio.fs.WindowsPath.parse(WindowsPath.java:92)
	at java.base/sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:229)
	at jdk.compiler/com.sun.tools.javac.file.FSInfo.getJarClassPath(FSInfo.java:112)
	at jdk.compiler/com.sun.tools.javac.file.CacheFSInfo.getJarClassPath(CacheFSInfo.java:93)
	at jdk.compiler/com.sun.tools.javac.file.Locations$SearchPath.addJarClassPath(Locations.java:423)
	at jdk.compiler/com.sun.tools.javac.file.Locations$SearchPath.addFile(Locations.java:413)
	at jdk.compiler/com.sun.tools.javac.file.Locations$SearchPath.addFiles(Locations.java:345)
	at jdk.compiler/com.sun.tools.javac.file.Locations$SearchPath.addFiles(Locations.java:333)
	at jdk.compiler/com.sun.tools.javac.file.Locations$SearchPath.addFiles(Locations.java:339)
	at jdk.compiler/com.sun.tools.javac.file.Locations$ClassPathLocationHandler.computePath(Locations.java:779)
	at jdk.compiler/com.sun.tools.javac.file.Locations$SimpleLocationHandler.setPaths(Locations.java:721)
	at jdk.compiler/com.sun.tools.javac.file.Locations.setLocation(Locations.java:2098)
	at jdk.compiler/com.sun.tools.javac.file.JavacFileManager.setLocation(JavacFileManager.java:927)
	at org.kie.memorycompiler.jdknative.NativeJavaCompiler.compile(NativeJavaCompiler.java:80)
	at org.kie.memorycompiler.AbstractJavaCompiler.compile(AbstractJavaCompiler.java:41)
	at org.kie.memorycompiler.KieMemoryCompiler.compileNoLoad(KieMemoryCompiler.java:130)
	at org.kie.memorycompiler.KieMemoryCompiler.compileNoLoad(KieMemoryCompiler.java:102)
	at org.kie.memorycompiler.KieMemoryCompiler.compile(KieMemoryCompiler.java:53)
	at org.kie.memorycompiler.KieMemoryCompiler.compile(KieMemoryCompiler.java:39)
	at org.drools.ancompiler.KieBaseUpdaterANC.inMemoryUpdate(KieBaseUpdaterANC.java:66)
	at org.drools.ancompiler.KieBaseUpdaterANC.run(KieBaseUpdaterANC.java:52)
	at org.drools.ancompiler.KieBaseUpdaterANC.generateAndSetInMemoryANC(KieBaseUpdaterANC.java:99)
	at org.optaplanner.core.impl.score.stream.drools.DroolsConstraintSessionFactory.buildKieBaseFromModel(DroolsConstraintSessionFactory.java:85)
	at org.optaplanner.core.impl.score.stream.drools.DroolsConstraintSessionFactory.buildSession(DroolsConstraintSessionFactory.java:126)
	at org.optaplanner.core.impl.score.director.stream.DroolsConstraintStreamScoreDirectorFactory.newConstraintStreamingSession(DroolsConstraintStreamScoreDirectorFactory.java:50)
	at org.optaplanner.core.impl.score.director.stream.DroolsConstraintStreamScoreDirector.resetConstraintStreamingSession(DroolsConstraintStreamScoreDirector.java:74)
	at org.optaplanner.core.impl.score.director.stream.DroolsConstraintStreamScoreDirector.setWorkingSolution(DroolsConstraintStreamScoreDirector.java:66)
	at org.optaplanner.core.impl.score.DefaultScoreManager.updateScore(DefaultScoreManager.java:45)
	at org.acme.schooltimetabling.rest.TimeTableResource.getTimeTable(TimeTableResource.java:66)
	at org.acme.schooltimetabling.rest.TimeTableResource_Subclass.getTimeTable$$superaccessor5(TimeTableResource_Subclass.zig:836)
	at org.acme.schooltimetabling.rest.TimeTableResource_Subclass$$function$$5.apply(TimeTableResource_Subclass$$function$$5.zig:29)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:54)
	at io.quarkus.arc.runtime.devconsole.InvocationInterceptor.proceed(InvocationInterceptor.java:63)
	at io.quarkus.arc.runtime.devconsole.InvocationInterceptor.monitor(InvocationInterceptor.java:49)
	at io.quarkus.arc.runtime.devconsole.InvocationInterceptor_Bean.intercept(InvocationInterceptor_Bean.zig:521)
	at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:41)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:41)
	at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:32)
	at org.acme.schooltimetabling.rest.TimeTableResource_Subclass.getTimeTable(TimeTableResource_Subclass.zig:794)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:170)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:130)
	at org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:643)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:507)
	at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$2(ResourceMethodInvoker.java:457)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:459)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:419)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:393)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:68)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:492)
	... 18 more
	Suppressed: java.lang.NullPointerException
		at org.optaplanner.core.impl.score.director.stream.DroolsConstraintStreamScoreDirector.close(DroolsConstraintStreamScoreDirector.java:120)
		at org.optaplanner.core.impl.score.DefaultScoreManager.updateScore(DefaultScoreManager.java:44)
		... 44 more
```

**Comment by triceo** [comment]
@DeiEFLE There has been a CI issue on our side, unfortunately. That issue has caused the necessary fix for your problem to not make it into OptaPlanner 8.7, even though we have it and have had it for weeks now. The next release is OptaPlanner 8.8, due hopefully in the next week or two.

In the meantime, you should be able to verify the fix if you upgrade Drools in your project from 7.54.0.Final to 7.55.0.Final. (Which is the same change we will be making in OptaPlanner 8.8.)

**Comment by DeiEFLE** [comment]
Thanks for the detail information.

I updated the version of all drools dependencies on the dependency tree to 7.55

> implementation "org.drools:drools-canonical-model:7.55.0.Final"
> implementation "org.drools:drools-core-reflective:7.55.0.Final"
> implementation "org.drools:drools-model-compiler:7.55.0.Final"
> implementation "org.drools:drools-core:7.55.0.Final"
> implementation "org.drools:drools-compiler:7.55.0.Final"
> implementation "org.drools:drools-ecj:7.55.0.Final"
> implementation "org.drools:drools-alphanetwork-compiler:7.55.0.Final"
> implementation "org.drools:drools-mvel:7.55.0.Final"

but still the behavior stays the same.

Please beleave me, I double checked. But the dep tree is quite long - so I did not post it here as a quote.
Don't hesitate to ask, if you want to see it, anyway.

**Comment by triceo** [comment]
@Christopher-Chianelli Can I ask you to double-check? We've seen the issue fixed previously.

**Comment by DeiEFLE** [comment]
The error is still there and after removing everything matching the pattern

> ^(?!.*?drool.*?7\.54).*?\r\n

this is what remains on the dependency tree

> |    |    +--- org.drools:drools-canonical-model:7.54.0.Final -> 7.55.0.Final
> |    |    +--- org.drools:drools-model-compiler:7.54.0.Final -> 7.55.0.Final
> |    |    +--- org.drools:drools-core:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-alphanetwork-compiler:7.54.0.Final -> 7.55.0.Final
> |    |    +--- org.drools:drools-model-compiler:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-core:7.54.0.Final -> 7.55.0.Final (*)
> |    |         +--- org.drools:drools-core-reflective:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-compiler:7.54.0.Final -> 7.55.0.Final (*)
> |    +--- org.drools:drools-core:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-canonical-model:7.54.0.Final -> 7.55.0.Final
> |    |    +--- org.drools:drools-model-compiler:7.54.0.Final -> 7.55.0.Final
> |    |    +--- org.drools:drools-core:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-alphanetwork-compiler:7.54.0.Final -> 7.55.0.Final
> |    |    +--- org.drools:drools-model-compiler:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-core:7.54.0.Final -> 7.55.0.Final (*)
> |    |         +--- org.drools:drools-core-reflective:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-compiler:7.54.0.Final -> 7.55.0.Final (*)
> |    +--- org.drools:drools-core:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-canonical-model:7.54.0.Final -> 7.55.0.Final
> |    |    +--- org.drools:drools-model-compiler:7.54.0.Final -> 7.55.0.Final
> |    |    +--- org.drools:drools-core:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-core-dynamic:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-alphanetwork-compiler:7.54.0.Final -> 7.55.0.Final
> |    |    +--- org.drools:drools-model-compiler:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-core:7.54.0.Final -> 7.55.0.Final (*)
> |    |         +--- org.drools:drools-core-reflective:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-compiler:7.54.0.Final -> 7.55.0.Final (*)
> |    +--- org.drools:drools-core:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-canonical-model:7.54.0.Final -> 7.55.0.Final
> |    |    +--- org.drools:drools-model-compiler:7.54.0.Final -> 7.55.0.Final
> |    |    +--- org.drools:drools-core:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-core-dynamic:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-alphanetwork-compiler:7.54.0.Final -> 7.55.0.Final
> |    |    +--- org.drools:drools-model-compiler:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-core:7.54.0.Final -> 7.55.0.Final (*)
> |    |         +--- org.drools:drools-core-reflective:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-compiler:7.54.0.Final -> 7.55.0.Final (*)
> |    +--- org.drools:drools-core:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-canonical-model:7.54.0.Final -> 7.55.0.Final
> |    |    +--- org.drools:drools-model-compiler:7.54.0.Final -> 7.55.0.Final
> |    |    +--- org.drools:drools-core:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-core-dynamic:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-alphanetwork-compiler:7.54.0.Final -> 7.55.0.Final
> |    |    +--- org.drools:drools-model-compiler:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-core:7.54.0.Final -> 7.55.0.Final (*)
> |    |         +--- org.drools:drools-core-reflective:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-compiler:7.54.0.Final -> 7.55.0.Final (*)
> |    +--- org.drools:drools-core:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-canonical-model:7.54.0.Final -> 7.55.0.Final
> |    |    +--- org.drools:drools-model-compiler:7.54.0.Final -> 7.55.0.Final
> |    |    +--- org.drools:drools-core:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-core-dynamic:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-alphanetwork-compiler:7.54.0.Final -> 7.55.0.Final
> |    |    +--- org.drools:drools-model-compiler:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-core:7.54.0.Final -> 7.55.0.Final (*)
> |    |         +--- org.drools:drools-core-reflective:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-compiler:7.54.0.Final -> 7.55.0.Final (*)
> |    +--- org.drools:drools-core:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-canonical-model:7.54.0.Final -> 7.55.0.Final
> |    |    +--- org.drools:drools-model-compiler:7.54.0.Final -> 7.55.0.Final
> |    |    +--- org.drools:drools-core:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-alphanetwork-compiler:7.54.0.Final -> 7.55.0.Final
> |    |    +--- org.drools:drools-model-compiler:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-core:7.54.0.Final -> 7.55.0.Final (*)
> |    |         +--- org.drools:drools-core-reflective:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-compiler:7.54.0.Final -> 7.55.0.Final (*)
> |    +--- org.drools:drools-core:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-canonical-model:7.54.0.Final -> 7.55.0.Final
> |    |    +--- org.drools:drools-model-compiler:7.54.0.Final -> 7.55.0.Final
> |    |    +--- org.drools:drools-core:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-core-dynamic:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-alphanetwork-compiler:7.54.0.Final -> 7.55.0.Final
> |    |    +--- org.drools:drools-model-compiler:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-core:7.54.0.Final -> 7.55.0.Final (*)
> |    |         +--- org.drools:drools-core-reflective:7.54.0.Final -> 7.55.0.Final (*)
> |    |    +--- org.drools:drools-compiler:7.54.0.Final -> 7.55.0.Final (*)
> |    +--- org.drools:drools-core:7.54.0.Final -> 7.55.0.Final (*)

**Comment by triceo** [comment]
@DeiEFLE Thank you, we'll look into it. It was fixed at one point, perhaps it broke again for some reason? We'll see.

**Comment by Christopher-Chianelli** [comment]
@DeiEFLE Tried locally on both `stable` and `development` on Windows with latest Gradle, could not reproduce. Did you fetch the latest commits from the branches? Do you build drools/other project locally on your machine (which can override the jars)?

**Comment by DeiEFLE** [comment]
@Christopher-Chianelli I am so sorry I lost track and did not respond to your questions. Shame on me!

First:
Today I checked, if the issue is still there, and I can tell you, it's not.

Just to have answered your questions:
- I went with the stable branch.
- As stated in the initial post I did fetch the latest commits (But to be honest, I wouldn't believe myself if I were you...). But if I wouldn't have, the fix posted by triceo should have worked anyway, I suppose.
- No, there were no other projects on the workspace. And overwriting jars... As far as I know, that is not how gradle works when using different versions of the same jar.

Anyhow, thanks for your time and effort.
